### ISSUE TITLE
[Bug]: ChoiceGroup "selectedKey" controlled state not working as expected when dynamically loading a selectedKey from captured data

### ISSUE THREAD

**Issue by tandrasi** [microsoft/fluentui#24605]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
    OS: Windows 10 10.0.22000
    CPU: (16) x64 AMD Ryzen 7 3700X 8-Core Processor
    Memory: 15.29 GB / 31.93 GB

NPM Packages:
    @fluentui/react: 8.93.0

Browsers:
    Edge: Spartan (44.22000.120.0), Chromium (104.0.1293.70)
    Internet Explorer: 11.0.22000.120
```

### Are you reporting Accessibility issue?

No

### Reproduction

https://codepen.io/tandrasi/pen/YzLKYZd

### Bug Description

**NOTE:** this issue is not encountered on the "office-ui-fabric-react" repo. Seems to just happen on the "@fluent/ui" repo.

## Actual Behavior

1. Initial load of ChoiceGroup, the user selects an option (i.e. Yes/No);
2. State change (using useState and useEffect) triggers and necessary logic is fired correctly to set relevant option in a global context which tracks all fields and user input;
3. The component unmounts;
4. When re-mounted, logic correctly fires to retrieve said option value from the global context;
5. **unexpected** attempting to set the "selectedKey" dynamically via a useEffect does not work. At this time the control is already in the DOM and does not get updated.

**Works**
`<ChoiceGroup
      ...
      selectedKey={"Yes"}/>`

**Doesn't Work**
`const [selectedKey, setSelectedKey] = useState<string | undefined>();`

`useEffect(() => {
 ...
  if (Object.prototype.hasOwnProperty.call(dataContext.formData, props.name)) {
	  defaultOption = dataContext.formData[props.name] as string | undefined;
  }
  setSelectedKey(defaultOption);
}, []);`

`<ChoiceGroup
      ...
      selectedKey={selectedKey}/>`

## Expected Behavior
1. Initial load of ChoiceGroup, the user selects an option (i.e. Yes/No);
2. State change (using useState and useEffect) triggers and necessary logic is fired correctly to set relevant option in a global context which tracks all fields and user input;
3. The component unmounts;
4. When re-mounted, logic correctly fires to retrieve said option value from the global context;
5. **expected** "selectedKey" is dynamically set via the useEffect, when for example, its value is found in the global context.

### Requested priority

Normal

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by micahgodbolt** [comment]
Can you create a codepen that demonstrates this issue? aka.ms/fluentpen

**Comment by tandrasi** [comment]
Interestingly, we found this issue with the ComboBox too. 

**We stumbled upon a fix** though...completely unintended, but if you change the **typing** of the selectedKey state variable from:

`const [selectedKey, setSelectedKey] = React.useState<string | undefined>();`

to:

`const [selectedKey, setSelectedKey] = React.useState(null);`

It all works as expected. Obviously, the typing should not be causing this but hopefully will bring you guys to a solution faster.

Again, this happens with the ComboBox too and could potentially be elsewhere.
